### DISCUSSION TITLE
How to open nvim-tree in regular size during startup?

### DISCUSSION THREAD

**Discussion by Rid1FZ** [nvim-tree/nvim-tree.lua#3004] [Q&A] [answered]
When I try to open any directory using neovim, nvim-tree is fired up. This is intended. But it takes up the whole neovim window. How can I make nvim-tree opened in regular size instead?

Current Behaviour:
![Screenshot 2024-11-16 123107](https://github.com/user-attachments/assets/0eb6d8d0-a0b1-41be-a80d-1cdc75868adf)

Intended Behaviour:
![Screenshot 2024-11-16 123209](https://github.com/user-attachments/assets/5bc92d9c-d36b-4e89-8fc6-b359e13245d2)

**Comment by Rid1FZ** [comment, marked as answer]
Got the solution. Nvim-tree actually hijacks current window if it is a directory (like netrw does). The default behaviour of Nvim-tree is to occupy the whole window (I don't know why!). So firstly, we need to disable the behaviour of hijacking:
```lua
hijack_directories = {
	enable = false,
	auto_open = false,
},
```
Now to open Nvim-tree when current window/buffer is a directory, we will use an `autocommand`:
```lua
vim.api.nvim_create_autocmd({ "VimEnter" }, {
	callback = function(data)
		if not (vim.fn.isdirectory(data.file) == 1) then
			return
		end

		vim.cmd.cd(data.file)
		require("nvim-tree.api").tree.open()
	end,
})
```

**Reply by alex-courtis** [reply]
See [wiki Open At Startup](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup) for complete information and recipes.

**Reply by Rid1FZ** [reply]
Thanks. I actually tried all the recipes before and none of those worked. So I opened this discussion. After more research, I found that the recipes were not working because of the behaviour of `hijacking` of `nvim-tree`.

**Reply by alex-courtis** [reply]
I've added a note at the start of [wiki Open At Startup](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup)

I'd be most grateful if you reviewed / updated that and added your [Recipe](https://github.com/nvim-tree/nvim-tree.lua/wiki/Open-At-Startup#recipes)

**Reply by Rid1FZ** [reply]
I've added my recipe as an instruction after other recipies. Please review...

**Reply by alex-courtis** [reply]
Many thanks @Rid1FZ
